### PR TITLE
PP-8693 Return 402 for gateway errors

### PR DESCRIPTION
**Pull request by james-peacock-ho**
## WHAT YOU DID
- Connector currently returns 500 when there are gateway errors.
- When front end receives 500 from connector this implies the problem is with connector rather than with the PSP, and causes unnecessary alerts.
- Instead 402 will now be returned when there are gateway errors.
- A change will also be made to the front end so that it handles 402. Front end will display the same error page to the user as it does for 500.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/paymentprocessor/resource/CardResource.java [hunk 1]
```diff
@@ -50,8 +50,7 @@
 import java.util.Map;
 
 import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
-import static uk.gov.pay.connector.util.ResponseUtil.badRequestResponse;
-import static uk.gov.pay.connector.util.ResponseUtil.serviceErrorResponse;
+import static uk.gov.pay.connector.util.ResponseUtil.*;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/paymentprocessor/resource/CardResource.java]: Think the IDE setting needs to change. We don't use wildcard imports https://gds-way.cloudapps.digital/manuals/programming-languages/java.html#imports

File: src/main/java/uk/gov/pay/connector/util/ResponseUtil.java [hunk 2]
```diff
@@ -11,12 +11,7 @@
 
 import static java.lang.String.format;
 import static javax.ws.rs.core.Response.Status;
-import static javax.ws.rs.core.Response.Status.ACCEPTED;
-import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
-import static javax.ws.rs.core.Response.Status.CONFLICT;
-import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
-import static javax.ws.rs.core.Response.Status.NOT_FOUND;
-import static javax.ws.rs.core.Response.Status.OK;
+import static javax.ws.rs.core.Response.Status.*;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/util/ResponseUtil.java]: Same here for wild card imports

File: src/test/java/uk/gov/pay/connector/it/resources/worldpay/WorldpayCardResourceIT.java [hunk 3]
```diff
@@ -20,9 +20,7 @@
 import static java.lang.String.format;
 import static java.util.UUID.randomUUID;
 import static javax.ws.rs.core.HttpHeaders.COOKIE;
-import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
-import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
-import static javax.ws.rs.core.Response.Status.OK;
+import static javax.ws.rs.core.Response.Status.*;
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/it/resources/worldpay/WorldpayCardResourceIT.java]: Wildcard imports

**Review by kbottla — CHANGES_REQUESTED** [review]

**Comment by kbottla** [comment]
depends on https://github.com/alphagov/pay-frontend/pull/3171

File: src/main/java/uk/gov/pay/connector/paymentprocessor/resource/CardResource.java (hunk 1 above)
Comment (james-peacock-ho) [reply src/main/java/uk/gov/pay/connector/paymentprocessor/resource/CardResource.java]: thanks, have corrected preferences and made changes

File: src/test/java/uk/gov/pay/connector/it/resources/worldpay/WorldpayCardResourceIT.java (hunk 3 above)
Comment (james-peacock-ho) [reply src/test/java/uk/gov/pay/connector/it/resources/worldpay/WorldpayCardResourceIT.java]: done

File: src/main/java/uk/gov/pay/connector/util/ResponseUtil.java (hunk 2 above)
Comment (james-peacock-ho) [reply src/main/java/uk/gov/pay/connector/util/ResponseUtil.java]: done

**Review by kbottla — APPROVED** [review]
LGTM! We should also update API specs for all endpoints but can be done in a separate PR  https://github.com/alphagov/pay-connector/blob/master/src/main/java/uk/gov/pay/connector/paymentprocessor/resource/CardResource.java#L143

**Pull request merged**